src/main/java/in/org/projecteka/hdaf/link/discovery/Discovery.java

```diff
@@ -19,6 +38,37 @@ public Discovery(ClientRegistryClient client) {
                 .map(Transformer::to);
     }
 
+    public Mono<DiscoveryResponse> patientFor(String providerId, String patientId, String transactionId) {
```

Comment:
nic: If there is a request class to send providerId instead of string parameter, it will be easier extension.